server/routes/public.js

```diff
@@ -28,18 +28,47 @@ router.get('/login', async (req, res) => {
     res.send(user);
 });
 
-router.get('/create-user', async (req, res) => {
-    const firstName = req.query.firstName;
-    const lastName = req.query.lastName;
-    const type = req.query.type; //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
-    const studentID = req.query.studentID;
-    const email = req.query.email;
-    const password = req.query.password;
-    const phone = req.query.phone;
+router.get('/api/create-user', async (req, res) => {
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
 
-    const user = await accountHandler.createUserAccount(firstName, lastName, type, studentID, email, password, phone);
+        //The following joi verifications make sure the values inputted adhere to the format requirements
+        //specified in the migrations> create-worker-table and migrations> create-student-table.
 
-    res.send(user);
-});
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        firstName: Joi.string().min(1).max(20).required(),
+        lastName: Joi.string().min(1).max(20).required(),
+        studentId: Joi.number().positive().integer(),
+        workerId: Joi.number().positive().integer(),
+        userType: Joi.string().min(1).valid(['student', 'worker']),
+        type: Joi.string(),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+    }).xor('studentId', 'workerId'); // Either the studentId or the workerId must be specified (they both cannot be specified).
```

Comment:
Hi Nancy,
I would recommend removing the semicolon after line 49 or it will not execute line 50 as well.

Reply:
Nice catch! I've removed the semi-colon!